src/components/node.js

```diff
@@ -87,6 +88,8 @@ class Node extends React.Component {
    */
 
   shouldComponentUpdate = (nextProps) => {
+    if (nextProps.readOnly !== this.props.readOnly) return true
```

Comment:
I'd like to move it down in the event that someone wants to have full control and not actually force a re-render should the `readOnly` value change, since many cases would be able to optimize that re-render away if there is no nesting. I'd rather opt for that escape hatch to be given complete override power instead of guessing what they might want.